src/tribler/core/components/torrent_checker/torrent_checker/dataclasses.py

```diff
@@ -15,13 +16,23 @@
 HEALTH_FRESHNESS_SECONDS = 4 * HOUR  # Number of seconds before a torrent health is considered stale. Default: 4 hours
 
 
+class Source(IntEnum):
+    """ Source of the Torrent Health information.
+    """
+    NONE = 0
+    TRACKER = 1
+    DHT = 2
+    POPULARITY_COMMUNITY = 3
+
+
 @dataclass(order=True)
 class HealthInfo:
     infohash: bytes = field(repr=False)
     seeders: int = 0
     leechers: int = 0
     last_check: int = field(default_factory=lambda: int(time.time()))
     self_checked: bool = False
+    source: Source = Source.NONE
```

Comment:
I think we can add a separate `tracker` string field with a tracker's URL.

Reply:
Added